main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DefCompleter.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 Paul Butcher, Lukas Rønn
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.api.lsp.Index
+import ca.uwaterloo.flix.api.lsp.provider.completion.Completion.DefCompletion
+import ca.uwaterloo.flix.language.ast.TypedAst
+
+object DefCompleter extends Completer {
+  /**
+    * Returns a List of Completion for defs.
+    */
+  override def getCompletions(implicit context: CompletionContext, flix: Flix, index: Index, root: Option[TypedAst.Root], delta: DeltaContext): Iterable[DefCompletion] = {
+    if (root.isEmpty) {
+      return Nil
+    }
+
+    val word = context.word
+    val uri = context.uri
+
+    root.get.defs.values.filter(matchesDef(_, word, uri)).flatMap(decl => Option(Completion.DefCompletion(decl, context, flix)))
```

Comment:
Not sure if it is possible to just use a map instead of flatMap.. This way we remove the 'Option'.

Reply:
Yes, I can. 

But I'm unsure of the original purpose of having it as an option, so that's why I didn't change it.

Reply:
I think its senseless, so just change it.